components/board/genericlinux/pin_types.go

```diff
@@ -41,3 +33,61 @@ type NoBoardFoundError struct {
 func (err NoBoardFoundError) Error() string {
 	return fmt.Sprintf("could not determine %q model", err.modelName)
 }
+
+// PinDefinition describes a gpio pin on a linux board.
+type PinDefinition struct {
+	Name            string `json:"name"`
+	Ngpio           int    `json:"ngpio"` // this is the ngpio number of the chip the pin is attached to
+	LineNumber      int    `json:"line_number"`
+	PwmChipSysfsDir string `json:"pwm_chip_sysfs_dir,omitempty"`
+	PwmID           int    `json:"pwm_id,omitempty"`
+}
+
+// PinDefinitions describes a list of pins on a linux board.
+type PinDefinitions struct {
+	Pins []PinDefinition `json:"pins"`
+}
+
+// UnmarshalJSON handles setting defaults for pin configs.
```

Comment:
This is right, but a little too brief. I'd explicitly mention that any numeric values omitted from the pin definition default to -1 instead of 0.